### ISSUE TITLE
[Failing Test] gce-master-scale-correctness: [sig-cli] Kubectl client Simple pod should support inline execution and attach

### ISSUE THREAD

**Issue by oxddr** [kubernetes/kubernetes#89811]
**Which jobs are failing**:
gce-master-scale-correctness

**Which test(s) are failing**:
[sig-cli] Kubectl client Simple pod should support inline execution and attach

**Since when has it been failing**:
Last fail on 04/02. Two more flakes in last month. 

**Testgrid link**:
https://testgrid.k8s.io/sig-scalability-gce#gce-master-scale-correctness

**Reason for failure**:
```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl/kubectl.go:539
Apr  2 12:45:27.953: Unexpected error:
    <exec.CodeExitError>: {
        Err: {
            s: "error running /workspace/kubernetes/platforms/linux/amd64/kubectl --server=https://35.231.117.228 --kubeconfig=/workspace/.kube/config --namespace=kubectl-3782 run run-test-2 --image=docker.io/library/busybox:1.29 --restart=OnFailure --attach=true --leave-stdin-open=true -- sh -c cat && echo 'stdin closed':\nCommand stdout:\n\nstderr:\nError from server (NotFound): pods \"run-test-2\" not found\n\nerror:\nexit status 1",
        },
        Code: 1,
    }
    error running /workspace/kubernetes/platforms/linux/amd64/kubectl --server=https://35.231.117.228 --kubeconfig=/workspace/.kube/config --namespace=kubectl-3782 run run-test-2 --image=docker.io/library/busybox:1.29 --restart=OnFailure --attach=true --leave-stdin-open=true -- sh -c cat && echo 'stdin closed':
    Command stdout:
    
    stderr:
    Error from server (NotFound): pods "run-test-2" not found
    
    error:
    exit status 1
occurred
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:606
```

**Anything else we need to know**:
\-

/sig scalability
/sig cli

**Comment by jkaniuk** [comment]
Additional fail 4 runs later, scalability correctness tests are mainly red recently due to kubectl flakes.

/priority critical-urgent
/assign @seans3

**Comment by zhouya0** [comment]
Feels like these failling tests have something to do with `stdin`.

**Comment by hasheddan** [comment]
Any updates here? @seans3 @jkaniuk

**Comment by liggitt** [comment]
https://storage.googleapis.com/k8s-gubernator/triage/index.html?pr=1&text=stdin%20closed&test=inline%20execution%20and%20attach#5385de90e734d4ebef9a

**Comment by soltysh** [comment]
I think the root-cause here will be similar to https://github.com/kubernetes/kubernetes/issues/87851#issuecomment-616623789.
/assign

**Comment by soltysh** [comment]
Probable fix in #90364

**Comment by soltysh** [comment]
@hasheddan this is similar to #87851 and I suspect that #90364 should fix it too. 
/close
